tools/create_class_definitions.py

```diff
@@ -1,24 +1,47 @@
+import argparse
 import collections
 import copy
 import json
 from pathlib import Path
 
 import construct
 
+parser = argparse.ArgumentParser()
+def game_argument_type(s: str) -> str:
+    if s not in ["dread", "sr"]:
+        raise ValueError(f"No enum named {s} found")
+    return s
+
+parser.add_argument("game", help="The game to create the class definitions for",
+                     type=game_argument_type, choices=list(["dread", "sr"]))
+args = parser.parse_args()
```

Comment:
Please don't create and run an argument parser in the module root. Though this is a `tools` script, it makes that you can't ever import this file.

Reply:
That was horrible thing to do :).
args determine types_path -> game_types -> type_lib -> primitive_to_construct.
Had to remove the type hint for `all_types: dict[str, Any]` as it was depending on type_lib which is just passed as an argument to the constructor now.